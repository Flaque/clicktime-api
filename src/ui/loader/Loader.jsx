import React from 'react';

class Loader extends React.Component {
  render() {
    return (
      <div id="loader">
          <div className="loader-content">
            <h4>{this.props.loaderMessage}</h4>
            <div className="nob pulse"></div>
          </div>
      </div>
    )
  }
}

export default Loader;
