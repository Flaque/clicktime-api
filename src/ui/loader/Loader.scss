@keyframes pulse {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.5);
  }
  100% {
    transform: scale(1);
  }
}

#loader {
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;

  .nob {
    width: 20px;
    height: 30px;
    background: white;
    border-radius: 50px; // Make as round as possible
    margin: auto;

    &.pulse {
      animation: pulse 1s infinite;
    }
  }
}
