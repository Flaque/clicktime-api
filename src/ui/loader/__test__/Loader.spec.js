import React from 'react';
import Loader from '../Loader.jsx';
import renderer from 'react-test-renderer';

test('Test that Loader is still the same', () => {
  const tree = renderer.create(
    <Loader loaderMessage="test load" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
