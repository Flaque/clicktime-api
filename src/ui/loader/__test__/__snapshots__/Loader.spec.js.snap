exports[`test Test that Loader is still the same 1`] = `
<div
  id="loader">
  <div
    className="loader-content">
    <h4>
      test load
    </h4>
    <div
      className="nob pulse" />
  </div>
</div>
`;
